1/main.py

```diff
@@ -0,0 +1,133 @@
+def shape(m):
+    assert m, "map must be not empty"
+    return len(m), len(m[0])
+
+
+def printer(ch, curr, max_index):
+    if curr == max_index - 1:
+        print(ch)
+    else:
+        print(ch, end='')
```

Comment:
Actually, there is no difference between ' and " in python. People usually try to stick to one of them inside one project, so I would advise you to pick one and use it everywhere.
`"I can't do it. Or can I?"`  <-- ' symbol inside " ", OK
`'It is a so-called "nexus event"'`  <-- " symbol inside ' ', OK
So, it's up to you, try to use one style while possible.